ksql-engine/src/main/java/io/confluent/ksql/planner/plan/AggregateNode.java

```diff
@@ -458,8 +445,25 @@ String getInternalColumnForExpression(final Expression expression) {
       return internalNameToIndexMap;
     }
 
-    private Map<String, String> getExpressionToInternalColumnNameMap() {
-      return expressionToInternalColumnNameMap;
+    private Expression resolveToInternal(final Expression exp) {
+      if (exp instanceof FunctionCall) {
+        final FunctionCall funcCall = (FunctionCall) exp;
+        final List<Expression> internalArgs = getInternalExpressionList(funcCall.getArguments());
+        return new FunctionCall(
+            funcCall.getLocation(), funcCall.getName(), funcCall.getWindow(), funcCall.isDistinct(),
+            internalArgs);
+      }
+
+      final String name = expressionToInternalColumnNameMap.get(exp.toString());
+      if (name != null) {
+        return new QualifiedNameReference(QualifiedName.of(name));
+      }
+
+      if (exp instanceof DereferenceExpression) {
+        throw new KsqlException("GROUP BY expression must be part of SELECT: " + exp.toString());
+      }
+
+      return exp;
```

Comment:
To support other kinds of expressions we would need to traverse the expression and map to internal column names. So shouldn't we raise an error in this case until we implement support for that? e.g. if a user does ```select count(*) group by foo + bar```

Reply:
I'll look to use a visitor in the follow up PR.

Reply:
Visitor pattern done.